src/overlays/actors/ovl_En_Am/z_en_am.c

```diff
@@ -89,81 +125,448 @@ static DamageTable D_808B10C8 = {
+        effectPos.x = randPlusMinusPoint5Scaled(65.0f) + this->actor.world.pos.x;
+        effectPos.y = randPlusMinusPoint5Scaled(10.0f) + (this->actor.world.pos.y + 40.0f);
+        effectPos.z = randPlusMinusPoint5Scaled(65.0f) + this->actor.world.pos.z;
+        EffectSsKiraKira_SpawnSmall(globalCtx, &effectPos, &sVelocity, &sAccel, &D_808B1118, &D_808B111C);
+    }
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_AMOS_WALK);
+    Actor_SpawnFloorDustRing(globalCtx, &this->actor, &this->actor.world.pos, 4.0f, 3, 8.0f, 0x12C, 0xF, 0);
+}
+
+void func_808AFF9C(EnAm* this) {
+    f32 lastFrame = Animation_GetLastFrame(&object_am_Anim_00033C);
+
+    Animation_Change(&this->skelAnime, &object_am_Anim_00033C, 0.0f, lastFrame, lastFrame, 0, 0.0f);
+    this->enemyCollider.info.bumper.dmgFlags = 0x80000088;
+    this->interactCollider.info.bumper.dmgFlags = 0x77CFFF77;
+    if (this->actor.colChkInfo.health != 0) {
+        this->enemyCollider.base.atFlags &= ~AT_ON;
+    }
+    Math_Vec3f_Copy(&this->actor.world.pos, &this->actor.home.pos);
+    this->actionFunc = func_808B0040;
+}
+
+void func_808B0040(EnAm* this, GlobalContext* globalCtx) {
+    if (((this->enemyCollider.base.ocFlags1 & OC1_HIT) && (this->enemyCollider.base.ocFlags2 & OC2_HIT_PLAYER)) ||
+        (this->interactCollider.base.acFlags & AC_HIT)) {
+        if (this->textureBlend == 0) {
+            func_808B00D8(this);
+        }
+    } else if (this->textureBlend > 10) {
+        this->textureBlend -= 10;
+    } else {
+        this->textureBlend = 0;
+        this->actor.flags &= ~1;
+        this->unk_23A = 0;
+    }
+}
+
+void func_808B00D8(EnAm* this) {
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_AMOS_WAVE);
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_AMOS_VOICE);
+    this->returnHomeTimer = 300;
+    this->actionFunc = func_808B0124;
+}
+
+void func_808B0124(EnAm* this, GlobalContext* globalCtx) {
+    s32 tempTextureBlend;
+    f32 cos;
+    u8 pad;
+    f32 rand;
+    f32 sin;
+
+    if (this->textureBlend + 20 >= 255) {
+        this->textureBlend = 255;
+        this->actor.flags |= 1;
+        this->enemyCollider.info.bumper.dmgFlags = 0x81C2C788;
+        this->interactCollider.info.bumper.dmgFlags = 0x760D3877;
+        this->enemyCollider.base.atFlags |= AT_ON;
+        this->actor.shape.yOffset = 0.0f;
+        func_808B0358(this);
+    } else {
+        tempTextureBlend = this->textureBlend + 20;
+        rand = randPlusMinusPoint5Scaled(10.0f);
+        cos = Math_CosS(this->actor.shape.rot.y) * rand;
+        sin = Math_SinS(this->actor.shape.rot.y) * rand;
+        this->actor.world.pos.x = this->actor.home.pos.x + cos;
+        this->actor.world.pos.z = this->actor.home.pos.z + sin;
+        this->textureBlend = tempTextureBlend;
+    }
+}
+
+void func_808B0208(EnAm* this, GlobalContext* globalCtx) {
+    // If the armos is against a wall, rotate and turn away from it
+    if ((this->actor.speedXZ > 0.0f) && (this->actor.bgCheckFlags & 8)) {
```

Comment:
```suggestion
    if ((this->actor.speedXZ > 0.0f) && (this->actor.bgCheckFlags & BGCHECK_CHECK_ONE_FACE)) {
```
prob need formater for that one